src/systemctl/systemctl-show.c

```diff
@@ -2261,7 +2261,7 @@ static int get_unit_dbus_path_by_pid(
          * sends the numeric PID. */
 
         pidfd = pidfd_open(pid, 0);
-        if (pidfd < 0 && ERRNO_IS_NOT_SUPPORTED(errno))
+        if (pidfd < 0 && ERRNO_IS_NOT_SUPPORTED(errno) && !ERRNO_IS_PRIVILEGE(errno))
```

Comment:
I switched this to `PidRef` instead in #31472.

Reply:
Let's first merge this, otherwise, it is hard to backport the fix to stable.